static/src/server.ts

```diff
@@ -190,56 +244,92 @@ function getScatterTileSvg(
     apiRoot: CONFIG.apiRoot,
   };
 
-  return fetchScatterData(tileProp).then((chartData) => {
-    const svgContainer = document.createElement("div");
-    drawScatter(
-      chartData,
-      svgContainer,
-      SVG_HEIGHT,
-      null /* tooltipHtml */,
-      tileConfig.scatterTileSpec,
-      SVG_WIDTH
-    );
-    return svgContainer.querySelector("svg");
-  });
+  return fetchScatterData(tileProp)
+    .then((chartData) => {
+      const svgContainer = document.createElement("div");
+      drawScatter(
+        chartData,
+        svgContainer,
+        SVG_HEIGHT,
+        null /* tooltipHtml */,
+        tileConfig.scatterTileSpec,
+        SVG_WIDTH
+      );
+      return {
+        svg: getProcessedSvg(svgContainer.querySelector("svg")),
+        data_csv: scatterDataToCsv(
+          chartData.xStatVar.statVar,
+          chartData.xStatVar.denom,
+          chartData.yStatVar.statVar,
+          chartData.yStatVar.denom,
+          chartData.points
+        ),
+        srcs: getSources(chartData.sources),
+        title: getChartTitle(
+          tileConfig.title,
+          getScatterRS(tileProp, chartData)
+        ),
+        type: "SCATTER",
+      };
+    })
+    .catch(() => {
+      console.log("Failed to get scatter tile result for: " + id);
+      return null;
+    });
 }
 
-// Gets the svg element for a line tile
-function getLineTileSvg(
+// Gets the TileResult for a line tile
+function getLineTileResult(
+  id: string,
   tileConfig: TileConfig,
   place: NamedTypedPlace,
   statVarSpec: StatVarSpec[]
-): Promise<SVGSVGElement> {
+): Promise<TileResult> {
   const tileProp = {
     apiRoot: CONFIG.apiRoot,
-    id: DOM_ID,
+    id,
     place,
     statVarSpec,
     svgChartHeight: SVG_HEIGHT,
     svgChartWidth: SVG_WIDTH,
     title: tileConfig.title,
   };
-
-  return fetchLineData(tileProp).then((chartData) => {
-    drawLine(tileProp, chartData, null);
-    return document.querySelector("svg");
-  });
+  return fetchLineData(tileProp)
+    .then((chartData) => {
+      const tileContainer = document.createElement("div");
+      tileContainer.setAttribute("id", id);
+      document.getElementById(DOM_ID).appendChild(tileContainer);
+      drawLine(tileProp, chartData, null);
+      return {
+        svg: getProcessedSvg(tileContainer.querySelector("svg")),
+        data_csv: dataGroupsToCsv(chartData.dataGroup),
+        srcs: getSources(chartData.sources),
+        legend: chartData.dataGroup.map((dg) => dg.label || "A"),
```

Comment:
what is "A"?

Reply:
I'm not sure, but that's what's used for the legend text in a line chart: https://github.com/datacommonsorg/website/blob/master/static/js/chart/draw.ts#L1024-L1026